staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go

```diff
@@ -106,6 +106,11 @@ func IsFullyQualifiedDomainName(fldPath *field.Path, name string) field.ErrorLis
 	if len(strings.Split(name, ".")) < 2 {
 		return append(allErrors, field.Invalid(fldPath, name, "should be a domain with at least two segments separated by dots"))
 	}
+	for _, label := range strings.Split(name, ".") {
```

Comment:
I think you need to special-case the trailing '.' if present

Reply:
Please use something like "www.dns-sd.org." as the test case thockin suggested.
See http://www.dns-sd.org/TrailingDotsInDomainNames.html for reference.

Reply:
@thockin @cheftako sorry if I misunderstood, but the trailing dot already exists since before
https://github.com/kubernetes/kubernetes/blob/413bc1a1d238efb7c4ba9e3aac2c381c93295aec/staging/src/k8s.io/apimachinery/pkg/util/validation/validation_test.go#L554

Do you mean that we need to add another trailing dot test case?

Reply:
Nevermind.  I missed that we strip the trailing dot above.